Model/Transaction/Backfill.php

```diff
@@ -82,31 +85,30 @@ class Backfill
     protected $logger;
 
     /**
-     * @param ScopeConfigInterface $scopeConfig
-     * @param RequestInterface $request
      * @param TransactionFactory $transactionFactory
      * @param OrderFactory $orderFactory
      * @param RefundFactory $refundFactory
      * @param Logger $logger
+     * @param \Magento\Backend\Block\Template\Context $context
      * @param \Magento\Sales\Api\OrderRepositoryInterface $orderRepository
      * @param \Magento\Framework\Api\FilterBuilder $filterBuilder
      * @param \Magento\Framework\Api\Search\FilterGroupBuilder $filterGroupBuilder
      * @param \Magento\Framework\Api\SearchCriteriaBuilder $searchCriteriaBuilder
      */
     public function __construct(
-        ScopeConfigInterface $scopeConfig,
-        RequestInterface $request,
         TransactionFactory $transactionFactory,
         OrderFactory $orderFactory,
         RefundFactory $refundFactory,
         Logger $logger,
+        \Magento\Backend\Block\Template\Context $context,
```

Comment:
@ryanreeves-taxjar This dependency was causing a blank config screen and various CLI errors locally. When this dependency is injected, somewhere down the chain it initiates an instance that needs to know the state of the application and triggering the following errors:

```
[Magento\Framework\Exception\SessionException]
Area code not set: Area code must be set before starting a session.

[Magento\Framework\Exception\LocalizedException]
Area code is not set
```

It seems like this should only be injected within a block class. I would restore the original dependencies for `scopeConfig` and `request`. To get `storeManager`, use `Magento\Store\Model\StoreManagerInterface` like elsewhere in the codebase.

Reply:
![m2 3-cli](https://user-images.githubusercontent.com/44789510/77965994-749c7a80-729f-11ea-8b94-69b27c4cda7e.png)
![m2 3-admin](https://user-images.githubusercontent.com/44789510/77965998-76663e00-729f-11ea-839a-24936de93327.png)
![m2 2-cli](https://user-images.githubusercontent.com/44789510/77966004-77976b00-729f-11ea-8620-dff1b32c36d9.png)
![m2 1-cli](https://user-images.githubusercontent.com/44789510/77966005-77976b00-729f-11ea-9c71-a60a228e4ef5.png)
![m2 2-admin](https://user-images.githubusercontent.com/44789510/77966006-78300180-729f-11ea-8c2c-d46d4b98fd0d.png)
![m2 1-admin](https://user-images.githubusercontent.com/44789510/77966008-78c89800-729f-11ea-9ab8-c1ebd36c44c5.png)

@fastdivision I've been unable to reproduce the error in any of my testing.  We've also failed Magento code reviews before for not using context.  

I'll go ahead and back these changes out, but I think this is worth investigating more in the future.

Reply:
Thanks for investigating @ryanreeves-taxjar. Specific version was `2.2.4`. Let me know if you see any errors after compiling and then running `bin/magento -vvv`.